ts/nni_manager/common/training_service_v3.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to TrialJobMetric.data
+ **/
+export type Metric = string;
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to HyperParameters.value
+ **/
+export type Parameter = string;
+
+export interface EnvironmentInfo {
+    id: string;
+    // TODO
+}
+
+export interface TrainingServiceV3 {
+    /**
+     *  Invoked during experiment initialization.
+     *
+     *  It should verify the config and raise error if the specified training resource is not available.
+     *
+     *  It should not start daemon on worker machine.
+     *  If another component of the experiment failed to initialize,
+     *  the process will abort without invoking any clean up function.
+     **/
+    init(): Promise<void>;
+
+    /**
+     *  Invoked after init().
+     *
+     *  It is suggested to resolve the promise after all daemons initialized.
+     **/
+    start(): Promise<EnvironmentInfo[]>;
+
+    /**
+     *  Invoked after start().
+     *
+     *  Currently the max clean up time is hard coded to 60s.
+     *  If the returned promise does not resolve in time, the process will abort.
+     **/
+    stop(): Promise<void>;
+
+
+    /* Following methods are guaranteed to be invoked after start() and before stop(). */
+
+    /**
+     *  "Upload" a directory and assign a name.
+     *  The name is guaranteed not to contain special characters.
+     **/
+    uploadDirectory(directoryName: string, path: string): Promise<void>;
+
+    /**
+     *  Create a trial process in specified environment, using the uploaded directory as PWD.
+     *
+     *  Return trial ID on success.
+     *  Return null if the environment is not available.
+     **/
+    createTrial(environmentId: string, trialCommand: string, directoryName: string): Promise<string | null>;
```

Comment:
I have concern on `trialCommand`, what about distributed trial?

Reply:
That's a new feature and current version is aiming on compatibility and alignment with v2.

Reply:
yeah, understand, but maybe we should think about ddp feature beforehand. so that, the design would not change too much in the evolving of training service.